portfolio/src/main/webapp/script.js

```diff
@@ -260,25 +260,66 @@ async function initMap() {
         addMarker(event.latLng);
     });
 
+    displayMarkers();
+
+    var input = document.getElementById("place-input");
+    var searchBox = new google.maps.places.SearchBox(input);
+    map.controls[google.maps.ControlPosition.TOP_LEFT].push(input);
+
+    // Restrict the search result near to the current viewport
+    map.addListener("bounds_changed", function() {
+        searchBox.setBounds(map.getBounds());
+    });
+
+    // Place a new marker on the place that the user searches
+    searchBox.addListener("places_changed", function() {
+        var places = searchBox.getPlaces();
+        if (places.length === 0) {
+            return;
+        }
+
+        var bounds = new google.maps.LatLngBounds();
+        places.forEach(function(place) {
+            if (!place.geometry) {
+                return;
+            }
+            if (place.geometry.viewport) {
+                bounds.union(place.geometry.viewport);
+            } else {
+                bounds.extend(place.geometry.location);
+            }
+        });
+        map.fitBounds(bounds);
+    });
+}
+
+async function displayMarkers() {
+    markers = [];
+    displayMarker(40.4382538, -79.9201019);
```

Comment:
It's worth leaving a comment about what is located at these hard-coded coordinates, or putting them into constants. Sometimes, when we have "magic numbers" like this, it's preferred to keep them as a const near the top of the file with a meaningful name, e.g., `const CHEESECAKE_FACTORY_LAT = 40.438;` Then, the readers of your code have a better idea about how you came up with the number.